### PR TITLE
open src changes [front-end]

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
These changes begin to transition the client side code into a format that could be made into an NPM package. A separate PR will be made to turn the code into an npm package

File: components/context/SolanaAuthProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+);
+export const SolanaAuthProvider: FC<SolanaAuthProviderProps> = ({
+  children,
+}) => {
+  return (
+    <WalletConnectionProvider>
+      <WalletModalProvider>
+        <SolanaSignInProvider
+          domain="deolate.space"
```
Comment (alfonso-paella) [inline components/context/SolanaAuthProvider.tsx]: Should this be configurable? 

Also, there's a typo

File: functions/completeSolanaAuth.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    payload = util.decodeUTF8(payload);
+    const publicKey = base58.decode(pubkey);
+    signature = qptua(signature);
+
+    // verify that the bytes were signed witht the private key
+    if (!sign.detached.verify(payload, signature, publicKey))
+      throw new Error("invalid signature");
+
+    // TODO: Create the JWT token with Firebase and send to client
```
Comment (alfonso-paella) [inline functions/completeSolanaAuth.ts]: I think this TODO is no longer needed

File: functions/completeSolanaAuth.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    if (!dbNonce) throw new Error("Public Key not in DB");
+
+    const { nonce, domain } = parsePayload(payload);
+
+    // verify the payload
+    const constructedMessage = signInMessage(nonce, domain);
+
+    if (constructedMessage !== payload) throw new Error("Invalid payload");
+
+    // TODO: verify the domain (dynamically)
```
Comment (alfonso-paella) [inline functions/completeSolanaAuth.ts]: I think this TODO is no longer relevant

File: functions/completeSolanaAuth.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  );
+
+/**
+ * Function to parse the payload (msg) from the client and return the nonce and domain used to sign in
+ * @param pl payload sent from the client
+ * @returns Object -> the nonce and domain to check against the DB
+ */
+const parsePayload = (pl: string): { nonce: string; domain: string } => {
+  const nonce = pl.substring(pl.indexOf("id=") + 3);
+  // TODO: Parse the domain and verify that it is correct, else reject
```
Comment (alfonso-paella) [inline functions/completeSolanaAuth.ts]: Is this TODO still applicable?

**Review by alfonso-paella — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@alfonso-paella Removed the TODOs and moved the props from the `SolanaSiInProvider` to the `SolanaAuthProvider` that is used in `_app.tsx`

File: pages/_app.tsx [hunk 5]
```diff
@@ -11,9 +11,25 @@ import { firebaseClient } from "../utils/firebaseClient";
...
+
+const signOut = () => {
+  return auth.signOut();
+};
+
 const App: FC<AppProps> = ({ Component, pageProps }) => {
   return (
-    <SolanaAuthProvider>
+    <SolanaAuthProvider
+      domain="desolate.space"
```
Comment (alfonso-paella) [inline pages/_app.tsx:27]: WDYT of puttng put `<your domain goes here>` or something like that

Comment (ghost) [reply pages/_app.tsx:27]: I plan to make those changes as I see what needs to be done to turn it into a package, but happy to make then now too.

Comment (alfonso-paella) [reply pages/_app.tsx:27]: ah let's merge then

**Comment by alfonso-paella** [comment]
Actually, when testing the code, I had to change the import on pages/index.tsx

from 
`import { SignInWithSolana } from "../components/SignIn";`

to 
`import { SignInWithSolana } from "../components/SignIn/SolanaAuthButton";`

**Pull request merged**